controllers/articlesController.js

```diff
@@ -44,4 +45,23 @@ function getAllCommentsForAnArticle(req, res, next) {
 
 }
 
-module.exports = { getParticularArticle, getAllArticles, getAllCommentsForAnArticle }
\ No newline at end of file
+function postComment(req, res, next) {
+
+    const {article_id} = req.params
+    const { username, body }  = req.body
+
+    return getArticleFromDB(article_id).then(() => {
+
+        return postACommentToDB(article_id, username, body)
+
+        }).then((commentAdded) => {
+
+            res.status(201).send({commentAdded : commentAdded})
+
+        }).catch(next)
```

Comment:
Great work!
You could clean this code a little, because the query to post a comment uses _INSERT_, the column constraints set out in the seed file come into play.
- for the column _article_id_ and _author_ it is a foreign key referencing another table
- for the column _body_ it is _NOT NULL_ and must contain a value

These constraints will cause PSQL to reject the promise if they are not satisfied in the query, meaning that you don't need to `Promise.reject` in your own code 💡